levenshtein.go

```diff
@@ -41,6 +41,25 @@ func ComputeDistance(a, b string) int {
 	if len(s1) > len(s2) {
 		s1, s2 = s2, s1
 	}
+
+	// remove trailing identical runes.
+	for i := 0; i < len(s1); i++ {
+		if s1[len(s1)-1-i] != s2[len(s2)-1-i] {
+			s1 = s1[:len(s1)-i]
+			s2 = s2[:len(s2)-i]
+			break
+		}
+	}
+
+	// Remove leading identical runes.
+	for i := 0; i < len(s1); i++ {
+		if s1[i] != s2[i] {
+			s1 = s1[i:]
+			s2 = s2[i:]
+			break
+		}
+	}
```

Comment:
Can we add some tests and benchmarks with strings having long repeated trailing and leading characters? I want to confirm that the cost of removing the repeated characters is lesser than actually comparing them.

Reply:
In fact, the long_string test does this (by chance) for the trailing characters, but I will add a leading one, and some tests on random strings to be sure there is no regression for other cases.

    Simple/long_string-16    1847.5n ± 0%   208.2n ± 0%  -88.73% (p=0.000 n=20)

Reply:
Thanks!